Scrapper-Service/plugins/guide2research.py

```diff
@@ -0,0 +1,378 @@
+                self.push_todb(conf_data)
+                self.logger.info(f"Number of Top Conferences added: {i} ")
+            except Exception as e:
+                self.logger.error(
+                    f"Error while parsing page {link}, find full trace {e}")
+        for i, link in enumerate(all_conference_links):
+            try:
+                conf_data = self._parse_all_conference(link=link)
+                self.push_todb(conf_data)
+                self.logger.info(f"Number of All Conferences added: {i}")
+            except Exception as e:
+                self.logger.error(
+                    f"Error while parsing {link}, full trace {e}"
+                )
+
+    def _get_conferences_list(self, which: str) -> List[str]:
+        """
+        parses topconferences baseurl and returns -
+        - a list of available links
+        """
+        if which == 'all':
+            self._get_all_conf_pages(url=self.all_confs_base_address)
+            links = []
+            for link in self.all_conf_page_links:
+                content = self.get_page(
+                    qlink=link, debug_msg=f"Extracted links from {link}").content
+                links.append(self._parse_all_conference_base(content=content))
+            all_links = list(itertools.chain(*links))
+            return all_links
+        elif which == 'top':
+            content = self.get_page(
+                qlink=self.top_conf_address, debug_msg=f"Extracted links from{self.top_conf_address}").content
+            return self._parse_top_conf_base(content)
+
+    # Parsers for top conferences
+
+    def _parse_top_conf_base(self, content: bytes) -> List[str]:
+        """
+        Wrapper to parse the `/topconf` page and returns
+        a list of links for active conferences
+
+        Args:
+        content: bytes
+
+        Returns:
+        list of conferences: list
+        """
+        soup = bs(content, "html5lib")
+        div_el = soup.find("div", attrs={"id": "ajax_content"})
+        conf_tables = div_el.findAll("table")
+        for table in conf_tables:
+            try:
+                anchor = table.h4.a
+                if not anchor:
+                    pass
+                else:
+                    link = f"http://www.guide2research.com{anchor['href']}"
+                    yield link
+            except Exception as e:
+                PageParsingError(
+                    f"The following error occured while trying to parse an anchor element {e}")
+        self.logger.debug('Successfully scraped links of all top conferences')
+
+    def _parse_top_conference(self, link: str) -> Conference:
+        """
+        Parses individual top conference page and,
+        return Conference object
+
+        Args:
+        ---
+        link: str
+
+        Returns:
+        ---
+        Conference: object
+        """
+        self.logger.debug(f'Parsing {link}')
+        page = requests.get(link, allow_redirects=True)
```

Comment:
* Use the provided function, this will block the thread since no timeout is given. Use the given function

Reply:
* Make same change , anywhere you may have missed it. Also post a mongodb schema layout of conference captured by your scrapper , one from top conference , and one from general. Should give us the idea in case any change required.

Reply:
Hey, there's a catch with the `get_page`method provided, it works fine when the page/route is returned. In case of individual conferences listed on both `top conferences` and `all conferences` there's a redirect that's happening and it is not allowed by the provided method. In such a case, using `get_page` would not work as I wanted it to.

Reply:
I can change the request method to include timeout if that's okay